lib/events.js

```diff
@@ -128,39 +128,43 @@ EventEmitter.prototype.emit = function emit(type) {
 
 EventEmitter.prototype.addListener = function addListener(type, listener) {
   var m;
+  var events;
+  var existing;
 
-  if (!util.isFunction(listener))
+  if (typeof listener !== 'function')
     throw TypeError('listener must be a function');
 
-  if (!this._events)
-    this._events = {};
-
-  // To avoid recursion in the case that type === "newListener"! Before
-  // adding it to the listeners, first emit "newListener".
-  if (this._events.newListener)
-    this.emit('newListener', type,
-              util.isFunction(listener.listener) ?
-              listener.listener : listener);
+  if (!(events = this._events))
+    events = this._events = {};
+  else {
+    existing = events[type];
```

Comment:
No, it *is* defined there. I have the separate else there because of the code following this line (`events.newListener` can't possibly exist if `this._events` isn't set, so there's no need to check in that case).